### PR TITLE
Update to Bevy `0.10`

### PR DESCRIPTION
**Pull request by geieredgar**
Hello,
this updates the code base to bevy version `0.10` and should be ready to merge.

### CODE REVIEW

**Comment by rparrett** [comment]
Oops, I just did this migration too.

I have a couple notes from my efforts (was about half done migrating the examples when I thought to check here) that look like they could be useful in this PR:

1.
    There are a couple spots in the examples where 
    ```
    .add_startup_system_to_stage(StartupStage::PostStartup, whatever)
    ```
    is being abused to get a commands flush between startup systems that can now be replaced with stuff like

    ```
    .add_startup_systems((a, apply_system_buffers, b))
    ```
2.
    `hex_neighbors` and `mouse_to_tile` should use the new `viewport_to_world_2d` function.

**Comment by rparrett** [comment]
3. Bevy code post-[https://github.com/bevyengine/bevy/pull/6885](https://github.com/bevyengine/bevy/pull/6885) seems to prefer e.g.
    ```diff
    -        image_bind_groups: Res<'w, ImageBindGroups>,
    +        image_bind_groups: SystemParamItem<'w, '_, Self::Param>,
    ````
    Although I couldn't explain why (except that reusing `Self::Param` seems tidy.)

**Comment by geieredgar** [comment]
I have updated the code to bevy commit bevyengine/bevy@64b0f19 and incorporated all suggested changes.

**Comment by rparrett** [comment]
I opened a PR against your branch to fix a warning when using the `atlas` feature.

Other than that, for what it's worth, this looks good to me. Tested all the examples with/without `atlas`, and a few on wasm.

**Review by Trouv — COMMENTED** [review]
I think this is pretty close - though I admit some of the deeper rendering changes are over my head

File: examples/hex_neighbors.rs [hunk 1]
```diff
@@ -466,11 +443,9 @@ fn main() {
         .init_resource::<TileHandleHexCol>()
         .init_resource::<TileHandleHexRow>()
         .init_resource::<FontHandle>()
-        .add_startup_system(spawn_tilemap)
-        .add_startup_system_to_stage(StartupStage::PostStartup, spawn_tile_labels)
-        .add_startup_system_to_stage(StartupStage::PostStartup, spawn_map_type_label)
-        .add_system_to_stage(CoreStage::First, camera_movement)
-        .add_system_to_stage(CoreStage::First, update_cursor_pos.after(camera_movement))
+        .add_startup_systems((spawn_tilemap, apply_system_buffers).chain().before(spawn_tile_labels).before(spawn_map_type_label))
+        .add_startup_systems((spawn_tile_labels, spawn_map_type_label))
```
[the remark this answers is not in the document]
Comment (geieredgar) [reply examples/hex_neighbors.rs]: I' have decided to use a new `SpawnLabelsSet`, to be more explicit with dependencies between systems.

**Comment by rparrett** [comment]
It's a minor thing, but if we're adding another set, I personally think the behavior would be more obvious/self-documenting/robust if it's a `SpawnTilemapSet`.

```rust
.add_startup_systems((spawn_tilemap, apply_system_buffers).chain().in_set(SpawnTilemapSet))
.add_startup_systems((spawn_tile_labels, spawn_map_type_label).after(SpawnTilemapSet))
```

It seems more likely that we would add additional systems that rely on the Tilemap having been spawned than the other way around.

**Comment by geieredgar** [comment]
Ok, I've replaced `SpawnLabelsSet` with `SpawnTilemapSet` and fixed the redundant `.before()` call I missed earlier in the `hex_neighbor` example.

**Review by Trouv — APPROVED** [review]
LGTM. I also checked all the examples w/ default features and they seemed fine. Just expressing my approval, I can't merge it

**Review by Sierra100010 — COMMENTED** [review]
Good

**Review by pauwell — APPROVED** [review]

**Comment by hammackj** [comment]
Any eta on when this will get merged? Loving this map =)

**Comment by m0xsec** [comment]
+1 on supporting this being merged when possible. thanks all!

**Review by rparrett — APPROVED** [review]

**Review by m0xsec — APPROVED** [review]

**Review by bzm3r — APPROVED** [review]
Thanks everyone for your patience on this! I am going to speak to @StarArawn to give write-access to @rparrett

**Comment by HyperAlch** [comment]
When will update your crate to 0.10 now that this has been merged?

image

It still says 0.9

**Pull request merged**